File: 프로그래머스/lv1/76501. 음양 더하기/음양 더하기.js
function solution(absolutes, signs) {
    let sum = 0;
    for (let i = 0; i < absolutes.length; i++) {
        if (signs[i]) {
            sum += absolutes[i];
        } else {
            sum -= absolutes[i];
        }
    }
    return sum;
}

// 위 코드에서 solution 함수는 absolutes 배열과 signs 배열을 입력으로 받아서 해당 정수들의 합을 계산하고 반환합니다.
// 반복문을 이용하여 absolutes 배열의 각 요소에 대해 signs 배열의 값을 확인하여 양수인 경우에는 더하고 음수인 경우에는 빼주어 합을 계산합니다.

File: 프로그래머스/lv1/76501. 음양 더하기/음양 더하기.py
# https://1ets-just-do-it.tistory.com/39

'''
1차 시도한 코드

def solution(absolutes, signs):
    num = []
    for i in range(len(absolutes)):
        if signs[i] == 'false':
            num.append(-absolutes[i])
        else:
            num.append(absolutes[i])
    return sum(num)
'''

def solution(absolutes, signs):
    result = 0
    for num, sign in zip(absolutes, signs):
        if sign == True:
            result += num
        else:
            result -= num
    return result